packages/react-utilities/src/hooks/useCallbackRef.ts

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import { useIsomorphicLayoutEffect } from './useIsomorphicLayoutEffect';
+
+/**
+ * Creates a MutableRef with ref change callback. Is useful as React.useRef() doesn't notify you when its content
+ * changes and mutating the .current property doesn't cause a re-render. An opt-out will be use a callback ref via
+ * React.useState(), but it will cause re-renders always.
+ *
+ * @param initialValue - initial ref value
+ * @param callback - a callback to run when value changes
+ * @param skipInitialResolve - a flag to skip an initial ref report
+ *
+ * @example
+ * const ref = useCallbackRef(0, (newValue, oldValue) =\> console.log(oldValue, '-\>', newValue);
+ * ref.current = 1;
+ * // prints 0 -\> 1
+ *
+ * @see https://reactjs.org/docs/hooks-reference.html#useref
+ * @see https://github.com/theKashey/use-callback-ref#usecallbackref---to-replace-reactuseref
+ */
+export function useCallbackRef<T>(
+  initialValue: T | null,
+  callback: (newValue: T | null, lastValue: T | null) => void,
+  skipInitialResolve?: boolean,
+): React.MutableRefObject<T | null> {
+  const isFirst = React.useRef(true);
+  const [ref] = React.useState(() => ({
```

Comment:
Interesting, it seems that `useConst` fits there well, actually.

Off topic: Just to clarify, `useMemo()` can't be used there as React can accidentally clean memory ([react docs](https://reactjs.org/docs/hooks-reference.html#usememo)). Do you know that `.useRef` is safe in this case? If yes, it will be nice to reference it in JSDoc of `useConst` as I often see `.useState` usage in other packages instead...